numpy/_core/src/multiarray/array_method.h

```diff
@@ -55,6 +55,7 @@ typedef struct PyArrayMethodObject_tag {
     PyArrayMethod_StridedLoop *unaligned_strided_loop;
     PyArrayMethod_StridedLoop *unaligned_contiguous_loop;
     PyArrayMethod_StridedLoop *contiguous_indexed_loop;
+    void *aux_data;
```

Comment:
Beyond the fact that it should be renamed, I think, the below is only relevant for followup or maybe if splitting out.

Just FYI (but that should be split out): I think I would be OK to move this up (maybe very high) and if we make things public, include the existing ones up to, and including, flags.

The "methods/functions" should *not* be public, in order to ensure we can deprecate and replace them (it would be nice to add an accessors for some so that a non-NumPy library could use our functions).
(We could also move down `name`, I really added it for debugging only.)

After adding this, it could probably also simplify the legacy arraymethod as it currently (ab)uses aux-data for static data, effectively.

Reply:
OK, thanks! I now call it `static_data` and moved it just beyond `flags`. It is still on the method, which is most logical if we allow filling it via `fill_array_method_from_slots`. 

I can see the logic of something on `context` itself, though. But maybe we need to think through a bit where exactly it will be useful...